components/gitpod-db/src/typeorm/migration/1634894637934-AddSlugToProject.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import {MigrationInterface, QueryRunner} from "typeorm";
+import { columnExists } from "./helper/helper";
+
+export class AddSlugToProject1634894637934 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        if (!(await columnExists(queryRunner, "d_b_project", "slug"))) {
+            await queryRunner.query("ALTER TABLE d_b_project ADD COLUMN `slug` varchar(255) NULL");
+        }
+        if (await columnExists(queryRunner, "d_b_project", "slug")) {
```

Comment:
Should we do something for Gitlab repos here as well?

Reply:
You mean for the existing GitLab repos? [That's the problem that I ran into](https://github.com/gitpod-io/gitpod/issues/5847#issuecomment-950418026), hence why this PR doesn't address existing GitLab repos. Existing GitLab ones have the wrong format in the `name` column, and we shouldn't populate the `slug` column with these. I'm not sure how to populate existing ones with the correct slug. [Ideas are very welcome](https://github.com/gitpod-io/gitpod/issues/5847#issuecomment-950418026) 🙏🏽

Reply:
#5847 was [closed](https://github.com/gitpod-io/gitpod/issues/5847#event-5554292751) when this PR merged.

Regarding the strategy for https://github.com/gitpod-io/gitpod/issues/5847#issuecomment-950418026, my initial approach (perhaps naive) would have been to "lookup" the slug in GitLab by pulling in the repo details using the repo url. There would be a risk of collisions when we do that since I would expect that project slugs have to be unique within the team or user's project namespace.

Upon further reflection however, I'm inclined to go for option # 3 

> Leave the existing projects be. In the PR, existing GitLab projects will have NULL in the slug column, and the name column will be used instead, which is the current situation.

because:
1) breaking URLs is not a good idea and
2) I expect that we will end up offering a way to edit project slugs at some point in the future, and when we do that, should also offer redirect capabilities like people have come to expect (from GitHub).